code/headers/ssd1306_oled_buffered.hpp

```diff
@@ -10,52 +10,90 @@ namespace r2d2::display {
      * SSD1306 buffered interface for an oled
      * Implements hwlib::window to easily use text and drawing functions that
      * are already implemented. Extends from r2d2::display::ssd1306_i2c_c
+     * 
+     * The template parameters are used for the parent class.
      */
-    class ssd1306_oled_buffered_c : public ssd1306_i2c_c {
+    template<std::size_t Cursor_Count, uint8_t Display_Size_Width, uint8_t Display_Size_Height>
+    class ssd1306_oled_buffered_c : public ssd1306_i2c_c<Cursor_Count, Display_Size_Width, Display_Size_Height> {
     private:
         /**
          * The buffer with the pixel data
          * The first byte is used for the data-prefix that the display driver
          * requires to be sent when sending pixel data.
          */
-        uint8_t buffer[(width * height / 8) + 1] = {};
+        uint8_t buffer[(Display_Size_Width * Display_Size_Height / 8) + 1] = {};
 
     public:
         /**
          * Construct the display driver by providing the communication bus and
          * the address of the display.
          */
         ssd1306_oled_buffered_c(r2d2::i2c::i2c_bus_c &bus,
-                                const uint8_t &address);
+                                const uint8_t &address)
+                : ssd1306_i2c_c<Cursor_Count, Display_Size_Width, Display_Size_Height>(bus, address)  {
+            // set the command for writing to the screen
+            buffer[0] = this->ssd1306_data_prefix;
+
+            // write the initalisation sequence to the screen
+            bus.write(address, this->ssd1306_initialization,
+                    sizeof(this->ssd1306_initialization) / sizeof(uint8_t));
+        }
 
         /**
          * @brief Write a pixel to the screen
          *
-         * @param x
+         * @param x 
          * @param y
-         * @param data
+         * @param data Colour of the pixel to write. data > 0 will set the pixel
+         * Data <= 0 will clear the pixel
          */
-        void set_pixel(uint16_t x, uint16_t y, const uint16_t data) override;
+        void set_pixel(uint16_t x, uint16_t y, const uint16_t data) override {
+            // calculate the index of the pixel
+            uint16_t t_index = (x + (y / 8) * this->size.x) + 1;
 
+            // set or clear the pixel
+            if (data) {
+                buffer[t_index] |= (0x01 << (y % 8));
+            } else {
+                buffer[t_index] &= ~(0x01 << (y % 8));
+            }
+        }
         /**
          * This clears the display this overrides the default clear of hwlib
```

Comment:
missing comma between `This clears the display` and `this overrides the default clear of hwlib`